docs/site/migration/boot-scripts.md

```diff
@@ -6,8 +6,249 @@ sidebar: lb4_sidebar
+
+To perform the same logic defined in a LoopBack 3 boot script, you should create
+a corresponding observer in your LoopBack 4 application.
+
+LoopBack 4 has its own life cycles at runtime and has `boot`, `start`, `stop` as
+its transition states. Observer allows you to participate in the `start` state,
+which is equivalent to the LoopBack 3 custom boot script.
+
+_Note: We highly recommend you learn about
+[Life cycle events and observers](https://loopback.io/doc/en/lb4/Life-cycle.html)
+first before reading the following guide._
+
+Take the two scripts(`server/boot/print-info.js` and
+`server/boot/create-sample.js`) above as examples, let's first migrate
+`server/boot/print-info.js` to your LoopBack 4 application.
+
+#### Creating Observer
+
+Run command `lb4 observer` to create an observer called `printInfo`:
+
+```sh
+lb4 observer
+? Observer name: printInfo
+? Observer group:
+   create src/observers/print-info.observer.ts
+   update src/observers/index.ts
+
+Observer PrintInfo was created in src/observers/
+```
+
+You will find a file called `PrintInfo.observer.ts` created in folder
+`src/observers/`. Open that file, there is a class called `PrintInfoObserver`,
+which implements `LifeCycleObserver` and has `start` and `stop` functions. The
+`start` function is where you should copy over the content of LB3's `printInfo`
+function.
+
+#### Injecting Application
+
+The LoopBack 3 boot script takes in `server` as its first parameter, which is
+essentially the `application` injected using key
+`CoreBindings.APPLICATION_INSTANCE`:
+
+```ts
+// ... imports
+export class PrintInfoObserver implements LifeCycleObserver {
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE) private app: Application,
+  ) {
+    // ... other class members
+  }
+```
+
+Now within the class `PrintInfoObserver` you have access to the application.
+
+#### Migrating Artifacts
+
+A boot script usually accesses or manipulates an application's artifacts like
+models, datasources, etc... So when moving its content to the `start` function,
+make sure you edit the code to retrieve those artifacts in LoopBack 4's way.
+
+For the print info script, since it retrieves a datasource's info, you need to
+**MIGRATE** the LoopBack datasource `db` to your LoopBack 4 application. The
+steps are well documented on page [migration/datasources.md](./datasources.md).
+And in the `start` function, **EDIT CODE** to retrieve the datasource using
+`this.app.getSync('datasources.db')`:
+
+_The code snippet is the complete observer file, the LB3 boot script logic is
+only in the `start` function_
+
+```ts
+import {
+  /* inject, Application, CoreBindings, */
+  lifeCycleObserver, // The decorator
+  LifeCycleObserver,
+  inject,
+  CoreBindings,
+  Application, // The interface
+} from '@loopback/core';
+import {juggler} from '@loopback/repository';
+
+/**
+ * This class will be bound to the application as a `LifeCycleObserver` during
+ * `boot`
+ */
+@lifeCycleObserver('')
```

Comment:
Can't the `''` be omitted?

Reply:
yep, removed and added one more section for setting the scripts order.